ui/src/app/streams/streams/streams.component.ts

```diff
@@ -458,7 +459,15 @@ export class StreamsComponent implements OnInit, OnDestroy {
   loadStreamMetrics() {
     const streamNames = this.streamDefinitions && Array.isArray(this.streamDefinitions.items) ?
       this.streamDefinitions.items
-        .filter(i => (i.status === 'deployed') || (i.status === 'deploying') || (i.status === 'partial'))
+        .filter(i => {
```

Comment:
This will still load stream metrics per page. What we need is to have this invoked only when the user clicks on the `show graph` link.

Reply:
```if (streamNames.length) {```
This statement insure that we don't call the WS if there is no streams in the filtered array.

Reply:
ok, I understand it better now. Thanks @oodamien!